src/ert/scheduler/lsf_driver.py

```diff
@@ -107,19 +108,24 @@ async def submit(
             stderr=asyncio.subprocess.PIPE,
         )
         stdout, stderr = await process.communicate()
-        try:
-            job_id = (
-                stdout.decode("utf-8")
-                .strip()
-                .replace("<", "")
-                .replace(">", "")
-                .split()[1]
-            )
-        except IndexError as err:
+        if process.returncode:
             logger.error(
-                f"Command \"{' '.join(bsub_with_args)}\" failed with error message: {stderr.decode()}"
+                f"Command \"{' '.join(bsub_with_args)}\" failed with "
+                f"returncode {process.returncode} and error message: {stderr.decode() or '<empty>'}"
+            )
+            raise RuntimeError(stderr.decode())
+
+        try:
+            stdout_decoded = stdout.decode()
+        except UnicodeDecodeError as err:
+            raise RuntimeError("Could not understand binary output from bsub") from err
+
+        match = re.search("Job <([0-9]+)> is submitted to .+ queue", stdout_decoded)
+        if match is None:
+            raise RuntimeError(
+                f"Could not understand '{repr(stdout.decode())}' from bsub"
```

Comment:
Nitpick: If you're going to have `stdout_decoded`, you should use it here as well.

```suggestion
                f"Could not understand '{repr(stdout_decoded)}' from bsub"
```

Reply:
nit picked.